### DISCUSSION TITLE
space-x-? does not work

### DISCUSSION THREAD

**Discussion by BITRU** [unocss/unocss#1054] [Q&A]
Tryed new space-x-4 class. But it does not work. the same code work on tailwind project, but does not in UnoCSS.

` <div class="flex space-x-4">
        <div>01</div>
        <div>02</div>
        <div>03</div>
      </div>`

it generates CSS
`space-x-3>:not([hidden])~:not([hidden])>:not([hidden])~:not([hidden]){--un-space-x-reverse:0;margin-left:calc(0.75rem * calc(1 - var(--un-space-x-reverse)));margin-right:calc(0.75rem * var(--un-space-x-reverse));}`

but we missing --un-space-x-reverse.  how to fix that?

**Comment by BITRU** [comment]
I think this is issue with unocss

from uno: 
`.space-x-3 >: not([hidden])~: not([hidden])
    >: not([hidden])~: not([hidden])
{
    --un-space-x-reverse: 0;
    margin - left: calc(0.75rem * calc(1 - var(--un-space-x-reverse)));
    margin - right: calc(0.75rem * var(--un-space-x-reverse));
}`

from tailwind:
`.space-x-3 > : not([hidden]) ~ : not([hidden]){
    --tw-space-x-reverse: 0!important;
    margin - right: calc(0.75rem * var(--tw-space-x-reverse)) !important;
    margin - left: calc(0.75rem * calc(1 - var(--tw-space-x-reverse))) !important;
}`

Extra >: not([hidden])~: not([hidden]) in uno

**Reply by chu121su12** [reply]
Could you please double-check your config?

[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIBGBzAtAdwAswAXAUykNwDMBXOOdAPgCgopRJZEUNq4yADyhIADghhlcg3ABZmbdh3DR4yNFjwAnMiFwA2AAyHmhgIzAA9CtZLlXNb024deoyaaGATFZuL2nKo8GjguugbGpgDMvpC2HNZxLLEQTEA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiQYAHpFjpsAQy4AbeJhz4iJYKQAUyVFDO9ABccADaVKhhehJcdDA6ALSBThh65IhwUk7QoXq0aHpwagC6VCWydAAW0DCxMHSh-qj6sfFJ1bX1enkwVjCJTqRwfZYD0Pa4pBhwNQBuGFDBowN99k5mqGqyIgqN4ZFyokoqPgCUssLyjCxsnDx8-OeXRwqS0n6HqHQS6xihAEYAHQAJheqAkaFweSqrDAjQA9Aj6CAgdUEZsWmoLuUkNikEA&options=N4IgzgLgTglgxhEAuaBXApgXyA) vs [Play](https://play.tailwindcss.com/INlH5oWmQp)

**Reply by BITRU** [reply]
I use [antfu](https://github.com/antfu)
/
[vitesse-nuxt3](https://github.com/antfu/vitesse-nuxt3)
And done clean setup. And have this issue.

maybe problem with the template.

**Reply by chu121su12** [reply]
There are 4 `not hidden`s in your example suggesting multiple space variant was applied.

**Reply by BITRU** [reply]
Upgrade UnoCSS to 0.37.4 fix this issue. Thank you
